src/opaque.rs

```diff
@@ -28,39 +34,25 @@ use zeroize::Zeroize;
 
 /// The message sent by the client to the server, to initiate registration
 pub struct RegisterFirstMessage<Grp> {
+    /// User identity
+    id_u: Vec<u8>,
     /// blinded password information
     alpha: Grp,
 }
 
 impl<Grp: Group> TryFrom<&[u8]> for RegisterFirstMessage<Grp> {
     type Error = ProtocolError;
     fn try_from(first_message_bytes: &[u8]) -> Result<Self, Self::Error> {
-        let checked_slice = check_slice_size(
-            first_message_bytes,
-            Grp::ElemLen::to_usize(),
-            "first_message_bytes",
-        )?;
-        // Check that the message is actually containing an element of the
-        // correct subgroup
-        let arr = GenericArray::from_slice(checked_slice);
-        let alpha = Grp::from_element_slice(arr)?;
-        Ok(Self { alpha })
-    }
-}
-
-impl<Grp: Group> RegisterFirstMessage<Grp> {
-    /// byte representation for the registration request
-    fn to_bytes(&self) -> GenericArray<u8, Grp::ElemLen> {
-        self.alpha.to_arr()
```

Comment:
Why lose this one?

Reply:
I think we should simply disallow a way to get bytes without calling serialize()